File: src/main/scala/server/Mode.scala
package server

/**
  * Application mode, either `Dev`, `Test`, or `Prod`.
  *
  * @see [[server.Mode]]
  */
sealed abstract class Mode(val asJava: utils.Mode)

object Mode {

  case object Dev extends server.Mode(utils.Mode.DEV)

  case object Test extends server.Mode(utils.Mode.TEST)

  case object Prod extends server.Mode(utils.Mode.PROD)

  lazy val values: Set[server.Mode] = Set(Dev, Test, Prod)
}

File: src/main/scala/server/utils/Mode.java
package server.utils;

/** Application mode, either `DEV`, `TEST`, or `PROD`. */
public enum Mode {
    DEV,
    TEST,
    PROD;

    public server.Mode asScala() {
        if (this == DEV) {
            return server.Mode.Dev$.MODULE$;
        } else if (this == PROD) {
            return server.Mode.Prod$.MODULE$;
        }
        return server.Mode.Test$.MODULE$;
    }
}
